services-camel/src/test/java/edu/unc/lib/dl/services/camel/images/AddDerivativeProcessorTest.java

```diff
@@ -109,41 +109,11 @@ public void init() throws Exception {
 
     @Test
     public void createEnhancementTest() throws Exception {
-
-        when(repoObjLoader.getBinaryObject(any(PID.class))).thenReturn(binary);
-        when(binary.getParent()).thenReturn(parent);
-        when(message.getBody()).thenReturn(result);
+        mvFile = moveDir.newFile(fileName + ".PNG");
+        mvFile.deleteOnExit();
 
         processor.process(exchange);
 
-        verify(parent).addDerivative(eq(slug), any(InputStream.class), eq(extensionlessName),
-                eq("image/png"), eq(PcdmUse.ThumbnailImage));
-    }
-
-    @Test
-    public void createEnhancementRetryTest() throws Exception {
-
-        when(binary.getParent())
-                .thenThrow(new MockitoException("Can't add derivative"))
-                .thenReturn(parent);
-
-        processor.process(exchange);
-
-        verify(binary, times(2)).getParent();
-        verify(parent).addDerivative(eq(slug), any(InputStream.class), eq(extensionlessName),
-                eq("image/png"), eq(PcdmUse.ThumbnailImage));
-    }
-
-    @Test(expected = RuntimeException.class)
-    public void createEnhancementRetryFailTest() throws Exception {
-
-        when(binary.getParent())
-                .thenThrow(new RuntimeException());
-
-        try {
-            processor.process(exchange);
-        } finally {
-            verify(binary, times(maxRetries + 1)).getParent();
-        }
+        assertTrue(mvFile.exists());
```

Comment:
After stepping through the test, it looks like this is testing that `<mvDir>/small_thumb.PNG` exists.
The file that should be created by the processor is:
`<mvDir>/derivative.PNG`
So it should be verifying that path exists instead

Reply:
Yes, this one should be good now